lib/lab_manager/provider/vsphere.rb

```diff
@@ -118,9 +128,10 @@ def create_vm(opts = {})
             'wait'          => true
           )
 
-          fail CreateVMError, "CreationFailed, retrying (#{vm_name})" unless machine['vm_ref']
+          fail CreateVMError, "CreationFailed, retrying (#{vm_name})" unless machine || machine['vm_ref']
```

Comment:
bad news for religious people, right?

I would suggest: "creation of (#{vm_name}) machine failed, retrying"